locals.tf

```
@@ -16,6 +16,10 @@ locals {
         password = base64encode(random_password.db_password.0.result)
         host     = "keycloak-postgres-db"
       }
+      pvc = {
+        enabled = false
+        size    = "1Gi"
+      }
```

```suggestion
        host     = "keycloak-postgres-db"
        pvc = {
          enabled = var.enable_postgres_pvc
          size    = "1Gi"
        }
      }
```

As I said in another comment, I think the the `pvc` values should be inside the `database` values. Also, I think the PVC should be enabled through a Terraform variable. Could you please create one?